app/src/main/kotlin/com/wire/android/ui/home/conversations/usecase/GetImageAssetMessagesFromConversationUseCase.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.ui.home.conversations.usecase
+
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
+import androidx.paging.insertSeparators
+import androidx.paging.map
+import com.wire.android.mapper.UIAssetMapper
+import com.wire.android.ui.common.monthYearHeader
+import com.wire.android.ui.home.conversations.model.messagetypes.asset.UIAssetMessage
+import com.wire.android.util.dispatchers.DispatcherProvider
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.feature.asset.GetPaginatedFlowOfAssetImageMessageByConversationIdUseCase
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.map
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toLocalDateTime
+import javax.inject.Inject
+import kotlin.math.max
+
+class GetImageAssetMessagesFromConversationUseCase @Inject constructor(
+    private val getAssetMessages: GetPaginatedFlowOfAssetImageMessageByConversationIdUseCase,
+    private val assetMapper: UIAssetMapper,
+    private val dispatchers: DispatcherProvider
+) {
+
+    /**
+     * This operation observers image asset messages from a conversation
+     * @param conversationId The conversation ID that it will look for image asset messages in.
+     *
+     * @return A [PagingData<UIImageAssetPagingItem>>] containing [UIImageAssetPagingItem.Asset] and time [UIImageAssetPagingItem.Label].
+     */
+    suspend operator fun invoke(
+        conversationId: ConversationId,
+        initialOffset: Int
+    ): Flow<PagingData<UIImageAssetPagingItem>> {
+        val pagingConfig = PagingConfig(
+            pageSize = PAGE_SIZE,
+            prefetchDistance = PREFETCH_DISTANCE,
+            initialLoadSize = INITIAL_LOAD_SIZE
+        )
+
+        return getAssetMessages(
```

Comment:
suggestion: maybe add some tests so that we know that these separators are always added the correct way 🤔